### DISCUSSION TITLE
Building in production: The .env is not copied into the build folder

### DISCUSSION THREAD

**Discussion by LeCoupa** [adonisjs/core#1874] [Help] [answered]
It seems that this issue [has also been reported on StackOverflow also](https://stackoverflow.com/questions/64691733/adonis-5-build-did-not-copy-into-build-folder).

Building with `npm run build` does not copy the .env file at the root of the project into the build folder.

Thus, the following error appears with `node build/server.js`

> Exception: E_MISSING_ENV_VALUE: Missing environment variable "NODE_ENV"

I am using `"@adonisjs/core": "5.0.4-preview-rc-2"`

**Comment by thetutlage** [comment, marked as answer]
This is intentional since you are not supposed to use your development `.env` file in production. 

- There are some values that needs to be changed. For example: `DB_HOST`, `DB_NAME`, `DB_PASSWORD` and so on
- The `APP_KEY` should be strictly different in prod and development.

In other words, the `.env` file is not meant to be portable and hence there is no point in copying it to the `build` folder

**Reply by LeCoupa** [reply]
I think there is something I am missing. This is not my `.env` file for development but for production as I build Adonis on my remote server.

```
git fetch --all
npm install
npm run build
pm2 start build/server.js --name API
```

This means that now, I have to run this command `cp .env build/.env` before starting the process or it will not work.

**Reply by thetutlage** [reply]
> This means that now, I have to run this command cp .env build/.env before starting the process or it will not work

Yes, you have to. There is no standard way of defining environment variables. 

- You decided to do it via `.env` file inside the project root in your production server.
- One using Heroku or AWS will define them using their management console's.
- One using DO apps will define them using the app spec file.

In any ways, the step of providing environment variables in prod has to be explicit. I know earlier (before the recent release), we used to copy the `.env` file. But this can lead to **false positive** behaviour too

**Reply by LeCoupa** [reply]
Thank you @thetutlage. By the way, I have a few hours tomorrow to work on the documentation regarding deployment. Where can I find the repository?

EDIT: Just found it https://github.com/adonisjs/adonisjs.com/
